storage/src/test/java/tech/pegasys/teku/storage/client/CombinedChainDataClientTest.java

```diff
@@ -146,6 +150,79 @@ void getsEarliestAvailableBlobSidecarSlot() {
     assertThat(result).hasValue(UInt64.ONE);
   }
 
+  @Test
+  void getStateForBlockProduction_directsToStateAtSlotExact()
+      throws ExecutionException, InterruptedException {
+    final BeaconState state = dataStructureUtil.randomBeaconState(UInt64.valueOf(2));
+    final Optional<Bytes32> recentBlockRoot =
+        Optional.of(spec.getBlockRootAtSlot(state, UInt64.ONE));
+    when(recentChainData.getBlockRootInEffectBySlot(UInt64.ONE)).thenReturn(recentBlockRoot);
+    when(recentChainData.getStore()).thenReturn(store);
+    when(store.retrieveStateAtSlot(any()))
```

Comment:
nit: I would require ONE here, to be sure, we haven't requested any other slot

Reply:
it was just messy, but can...